src/Compilers/CSharp/Portable/Binder/Binder.cs

```diff
@@ -775,5 +776,73 @@ internal BoundStatement WrapWithVariablesAndLocalFunctionsIfAny(CSharpSyntaxNode
                                   ImmutableArray.Create(statement))
             { WasCompilerGenerated = true };
         }
+
+#if DEBUG
+        internal string Dump()
+        {
+            return TreeDumper.DumpCompact(this.DumpAncestors());
+        }
+
+        private TreeDumperNode DumpAncestors()
+        {
+            TreeDumperNode current = null;
+
+            for (var scope = this; scope != null; scope = scope.Next)
+            {
+                var (description, snippet, locals) = Print(scope);
+                var sub = new List<TreeDumperNode>();
+                if (!locals.IsEmpty())
+                {
+                    sub.Add(new TreeDumperNode("locals", locals, null));
+                }
+                var currentContainer = scope.ContainingMemberOrLambda;
+                if (currentContainer != null && currentContainer != scope.Next?.ContainingMemberOrLambda)
+                {
+                    sub.Add(new TreeDumperNode("containing member or lambda", currentContainer.ToDisplayString(), null));
```

Comment:
>containing member or lambda [](start = 48, length = 27)

I would call it ContainingSymbol, effectively this is what it is. For example, a local function is neither a member, nor a lambda.